### PR TITLE
Prepare Well Implemantion For Water and Water-Gas System

### PR DESCRIPTION
**Pull request by goncalvesmachadoc**

### CODE REVIEW

File: opm/simulators/wells/StandardWell_impl.hpp [hunk 1]
```diff
@@ -256,30 +256,47 @@ namespace Opm
...
+            if (has_solvent && compIdx == (unsigned)contiSolventEqIdx) {
+                return primary_variables_evaluation_[SFrac];
+            }
         }
+        else if (FluidSystem::phaseIsActive(FluidSystem::waterPhaseIdx) && compIdx == Indices::canonicalToActiveComponentIndex(FluidSystem::waterCompIdx)) {
+             return primary_variables_evaluation_[WFrac];
 
-        if (has_solvent && compIdx == (unsigned)contiSolventEqIdx) {
-            return primary_variables_evaluation_[SFrac];
+             if (FluidSystem::phaseIsActive(FluidSystem::gasPhaseIdx) && compIdx == Indices::canonicalToActiveComponentIndex(FluidSystem::gasCompIdx)) {
```
Comment (totto82) [inline opm/simulators/wells/StandardWell_impl.hpp:275]: This part of the code seems to be after the return.

**Review by totto82 — CHANGES_REQUESTED** [review]
Thanks for the fix. Aside from comments issue I think this can be merged (if Jenkins agrees)

**Comment by totto82** [comment]
There is lot of ifs here (and elsewhere) Is there a better way to handle the flexibility of 1p, 2p, 3p systems? Maybe an iterator over a subset of the components?

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this please

**Comment by goncalvesmachadoc** [comment]
Hi @totto82 , is that fine? Can it be merged?

**Review by totto82 — APPROVED** [review]

**Comment by totto82** [comment]
Thanks. I just forgot.

**Pull request merged**